<a href="https://colab.research.google.com/github/MayraDemoss/Descriptiva/blob/main/Variaci%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 2. Dispersión, variabilidad o heterogeneidad

Estudiar la distribución de los valores de la serie, analizando si estos se encuentran más o menos concentrados, o más o menos dispersos. 

![tinystats.github.io](https://tinystats.github.io/teacups-giraffes-and-statistics/images/04_variance/giraffe_variance2.jpg)

***
> ¿Qué tan homogeneas son las exhalaciones del geyser Old Faithful?
***
Existen diversas medidas de dispersión, entre las más utilizadas podemos destacar las siguientes:

## Amplitud.
Es la medida de dispersión más fácil de obtener. Se obtiene hallando la diferencia entre la observación mayor y la menor (el valor máximo menos el valor mínimo). La amplitud es una medida muy influenciada por los valores extremos y por lo tanto puede dar una impresión falsa sobre los valores reales de la distribución.




In [ ]:
# La amplitud del tiempo que tardan las exhalaciones y el  tiempo entre exhalaciones
max(faithful$eruptions)-min(faithful$eruptions)   
max(faithful$waiting)-min(faithful$waiting)   

# la función range proporciona el mínimo y el máximo
range(faithful$eruptions)
range(faithful$waiting)

# diferencia
diff(range(faithful$eruptions))
diff(range(faithful$waiting))


[1] 3.5

[1] 53

[1] 1.6 5.1

[1] 43 96

[1] 3.5

[1] 53

Interpretación:

## Varianza.
Mide la distancia (cuadrada) existente entre los valores de la serie y la media. Mientras más se aproxima a cero, más concentrados están los valores de la serie alrededor de la media. Por el contrario, mientras mayor sea la varianza, más dispersos están. El símbolo para representar a la varianza muestral es
$s^2$ y para la varianza poblacional $\sigma^2$.

$$s^2=\frac{1}{n-1}{\sum_{i=1}^{n}\left(x_i-\bar{x}\right)^2}$$

$$\sigma^2=\frac{1}{N}{\sum_{i=1}^{N}\left(x_i-\mu\right)^2}$$

## Desviación estándar o típica.
Se calcula como raíz cuadrada de la varianza.

Nótese que ni la varianza ni la desviación estándar pueden ser negativas y son cero sólo cuando todos los datos tienen el mismo valor que no es otro que el valor de la media.


In [ ]:
# La varianza de las exhalaciones y el tiempo entre exhalaciones
var(faithful$eruptions)
var(faithful$waiting)

mean(faithful$eruptions)

[1] 1.302728

[1] 184.8233

[1] 3.487783

Interpretación: Los tiempos que tarda una exhalación se alejan en promedio 1.3 seg^2 de 3.49 seg


In [ ]:
# desviación estándar
sd(faithful$eruptions)
sd(faithful$waiting)


[1] 1.141371

[1] 13.59497

Los tiempos que tardan las exhalaciones se alejan en promedio 1.14 seg de 3.49 seg


* Se puede usar a la amplitud para estimar a la desviación estándar ([Range Rule for Standard Deviation](https://scholar.rose-hulman.edu/cgi/viewcontent.cgi?article=1211&context=rhumj)).

* [Estimación de la media y de la desviación estándar a partir de la amplitud](https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-14-135)

* [Use of the Range Instead of the Standard Deviation](https://www.jstor.org/stable/2281203).


In [ ]:
# Aproximaciones de la desviación estándar
# estimación de la desviación estándar
diff(range(faithful$eruptions)) / 4
diff(range(faithful$waiting)) / 4



[1] 0.875

[1] 13.25

In [ ]:
# aproximación de la S a partir de la amplitu en los datos de 
# glucosa
glucosa <- read.csv('glucosa.csv')

diff(range(glucosa$glucosamgpordL))/4
diff(range(glucosa$glucosamgpordL))/6

sd(glucosa$glucosamgpordL)




[1] 92.25

[1] 61.5

[1] 57.76527

# La [regla empirica](https://online.stat.psu.edu/stat200/lesson/2/2.2/2.2.7)
<center>

![](https://upload.wikimedia.org/wikipedia/commons/thumb/2/22/Empirical_rule_histogram.svg/553px-Empirical_rule_histogram.svg.png)

</center>

In [ ]:
# Regla empirica

mean(faithful$eruptions)-3*sd(faithful$eruptions)
mean(faithful$eruptions)+3*sd(faithful$eruptions)



[1] 0.06366933

[1] 6.911897

[1] 1.6 5.1

Más del 99% de las exhalaciones tienen tiempos de duración entre 0.06 y 6.9 seg


In [ ]:
# En total hay 272 datos
# dim(faithful)
empirico <- faithful[faithful$eruptions >= 0.06 & faithful$eruptions <= 6.9, ]
#empirico

272/272*100   # si se cumple la regla empirica

# sin generar un dataframe
faithful$eruptions >= 0.06 & faithful$eruptions <= 6.9

sum(faithful$eruptions >= 0.06 & faithful$eruptions <= 6.9) / 272

[1] 100

[1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [16] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [31] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [46] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [61] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [76] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [91] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[106] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[121] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[136] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[151] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[166] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[181] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[196] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[211] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[226] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[241] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[256] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[271] TRUE TRUE

[1] 1

In [ ]:
# otra forma
ecdf(faithful$eruptions)(6.91)-ecdf(faithful$eruptions)(.064)


[1] 1


## Amplitud intercuartil

Es una medida resistente de variación,puesto que elimina al 25% de datos más pequeños y al 25% de los datos mas grandes mide la amplitud del 50% de los datos centrales. 

$$AIC=P_{0.75}-P_{0.25}$$

## Desviación absoluta de la mediana

Es la mediana de las desviaciónes absolutas de cada valor respecto a la mediana.



In [ ]:
# amplitud intercuartil
quantile(faithful$eruptions,c(.75,.25))

abs(diff(quantile(faithful$eruptions,c(.75,.25))))


75%     25% 
4.45425 2.16275

25% 
2.2915

In [ ]:
median((abs(faithful$eruptions-median(faithful$eruptions))))

median((abs(faithful$waiting-median(faithful$waiting))))


[1] 0.6415

[1] 8

In [ ]:
# comparación de las medidas de variación
x = c(2, 4 ,8 ,9 ,11, 11, 12)

median((abs(x-median(x))))
var(x)
sd(x)
diff(quantile(x,c(.25,.75),type = 2))

y =c(2, 4, 8, 9, 11, 11, 120) 

median((abs(y-median(y))))
var(y)
sd(y)
diff(quantile(y,c(.25,.75),type = 2))


[1] 2

[1] 14.47619

[1] 3.804759

75% 
  7

[1] 2

[1] 1819.619

[1] 42.65699

75% 
  7

## Compartir Notebooks a través de GitHub

[GitHub](github.com) es una plataforma de desarrollo colaborativo para alojar proyectos utilizando el sistema de control de versiones Git. Se utiliza principalmente para la creación de código fuente de programas de ordenador. 

En nuestro caso lo usaremos como un repositorio que nos permitirá compartir nuestras Notebooks con código de R y para crear una página web personal.

## Compartir Notebooks a través de GitHub

[GitHub](github.com) es una plataforma de desarrollo colaborativo para alojar proyectos utilizando el sistema de control de versiones Git. Se utiliza principalmente para la creación de código fuente de programas de ordenador. 

En nuestro caso lo usaremos como un repositorio que nos permitirá compartir nuestras Notebooks con código de R y para crear una página web personal.
Mucio Osorio Sanchez19:38
[texto del vínculo](https://)Suponga que requiere compartir esta Notebook a través de GitHub.

1. Iniciar sesión en GitHub

![](https://github.com/mucioosorio/Diplomado-con-R.-Mayo-de-2021/blob/main/git1.JPG?raw=true)


3. Escribir el nombre del repositorio y seleccionar si el repositorio será publico o privado (opcionalmente agregar el README file). Luego oprimir el botón de crear repositorio.

![](https://github.com/mucioosorio/Diplomado-con-R.-Mayo-de-2021/blob/main/git3.JPG?raw=true)

Ejemplo de un repositorio público:

https://github.com/demar01/penguinsbox

5. Compartir la Notebook

Seleccionr en el menú principal la opción **Archivo** y luego **Guardar una copia en GitHub**

![](https://github.com/mucioosorio/Diplomado-con-R.-Mayo-de-2021/blob/main/git5.JPG?raw=true)